robot/impl/resource_manager.go

```diff
@@ -356,19 +371,36 @@ func (manager *resourceManager) internalResourceNames() []resource.Name {
 func (manager *resourceManager) ResourceNames() []resource.Name {
 	names := []resource.Name{}
 	for _, k := range manager.resources.Names() {
-		if k.API == client.RemoteAPI ||
-			k.API.Type.Namespace == resource.APINamespaceRDKInternal {
-			continue
+		if manager.includeResourceName(k) {
+			names = append(names, k)
 		}
-		gNode, ok := manager.resources.Node(k)
-		if !ok || !gNode.HasResource() {
-			continue
+	}
+	return names
+}
+
+// reachableResourceNames returns the names of all reachable resources in the manager.
+func (manager *resourceManager) reachableResourceNames() []resource.Name {
+	names := []resource.Name{}
+	for _, k := range manager.resources.ReachableNames() {
+		if manager.includeResourceName(k) {
+			names = append(names, k)
 		}
-		names = append(names, k)
 	}
 	return names
 }
 
+func (manager *resourceManager) includeResourceName(k resource.Name) bool {
+	if k.API == client.RemoteAPI ||
+		k.API.Type.Namespace == resource.APINamespaceRDKInternal {
+		return false
+	}
+	gNode, ok := manager.resources.Node(k)
+	if !ok || !gNode.HasResource() {
+		return false
+	}
+	return true
+}
```

Comment:
factored out to ensure `ResourceNames` and `reachableResourceNames` filter for the same resources. another approach instead of having multiple functions is to allow `ResourceNames` to receive an optional parameter that configures what kinds of resources are filtered  (i think @dgottlieb suggested a version of this approach)

Reply:
Ah sorry I got mixed up, you mentioned this for the proto not the method! My mistake - I meant to give credit not incept :)